src/data/nat/bits.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2022 Praneeth Kolichala. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Praneeth Kolichala
+-/
+import tactic.generalize_proofs
+import tactic.norm_num
+
+/-!
+# Additional properties of binary recursion on `nat`
+
+This file documents additional properties of binary recursion,
+which allows us to more easily work with operations which do depend
+on the number of leading zeros in the binary representation of `n`.
+For example, we can more easily work with `nat.bits` and `nat.size`.
+
+See also: `nat.bitwise`, `nat.pow` (for various lemmas about `size` and `shiftl`/`shiftr`),
+and `nat.digits`.
+-/
+
+namespace nat
+
+lemma bit_eq_zero_iff (n : ℕ) (b : bool) : bit b n = 0 ↔ n = 0 ∧ b = ff :=
```

Comment:
These can be implicit since they appear on both sides of the iff.
```suggestion
lemma bit_eq_zero_iff {n : ℕ} {b : bool} : bit b n = 0 ↔ n = 0 ∧ b = ff :=
```